### ISSUE TITLE
Code style / logging

### ISSUE THREAD

**Issue by baka-kaba** [Awful/Awful.apk#422]
I'm stripping out some of the crap I noticed while I was doing the forums index, just trying to pare down some of the other classes - removing unused code, refactoring some shared stuff. I'm removing a lot of the log spam, but it feels like I'm removing a ton of it.

I'm trying to strike a balance between taking out the pointless stuff you'd just add in if you really need it (like logging 'hello from onCreate()') and keeping in potentially useful information that might point someone in the right direction. Like in the WebChromeClient we have these:

```
@Override
public boolean onCreateWindow(WebView view, boolean isDialog,
		boolean isUserGesture, Message resultMsg) {
	if(DEBUG) Log.d(TAG,"onCreateWindow"+(isDialog?" isDialog":"")+(isUserGesture?" isUserGesture":""));
	return super.onCreateWindow(view, isDialog, isUserGesture, resultMsg);
}
```
which *might* be useful if you need to debug it in the future (and it doesn't log too often anyway), on the other hand that logging line is the only reason that method exists, and there are a bunch of those in a 1600+ line class file. As it is, it doesn't feel like it's logging useful information in general, and personally I'm leaning towards removing it until there's an actual issue. But some kind of consensus would be better

**Comment by baka-kaba** [comment]
As for code style, I've got into the habit of doing Ctrl+L in Android Studio and just letting it reformat my code before I commit it - it'll force a lot of whitespace changes in existing files though. I'd only do that kind of thing as a separate commit, but are we cool with that? I think it would help a ton if everyone's just using the defaults and doing a one-click cleanup, but I don't want to go forcing a new code style on the whole project

**Comment by Sereri** [comment]
Doing gods work I see.

I'd say all methods that only have a log and a super call can be removed.

As for code style, yeah just auto-format the thing. I've been holding off because of the same reasons but if it's a seperate commit it'll probably be ok. I'm usually fond of tabs instead of spaces but I'd go with what ever is used by default in Android Studio

**Comment by baka-kaba** [comment]
So am I cool aggressively removing things? Like there's logging that says it's restoring savedInstance state, which I think is *probably* fine in debug builds. When you load a thread you get spammed with URL requests, which is trickier because it's useful info, but I also don't generally care about all that - it's just more noise to scroll through. (Maybe comment that out?) Things like hearing what % the progress bar has updated to... I think that needs to go once it's working

Oh also I'm changing the logging levels, even though I used to do the 'use warn and error because they have different colours' thing, if that's ok

Auto-format is rad and now I don't care about tabs or spaces. It's refreshing

**Comment by Sereri** [comment]
> When you load a thread you get spammed with URL requests, which is trickier because it's useful info, but I also don't generally care about all that - it's just more noise to scroll through. (Maybe comment that out?)

Isn't that what the DEBUG check is for though?

And yeah, everything that isn't a crash shouldn't use Log.e

**Comment by baka-kaba** [comment]
Yeah but we're always in DEBUG now, because that's set to Constants.DEBUG and that's set by the build type. You think we should just set it to false and wrap all the verbose logging in a DEBUG check, so it's optional? (Maybe rename it to FULL_LOGGING or something)

Part of the reason I'm doing this is to strip out as much logging from the production build as I can, the less I/O the better

**Comment by baka-kaba** [comment]
I added the code style xml to #425 which doesn't have any actual changes to the default, it's just there so anything we actually change will propagate and everyone will have the same settings

I guess it's worth doing a Reformat Code on the entire module, and committing that? I just ran it and it seems fine, but that's over 200 files and it seems like it would be good to do it at a quiet time. Like when someone doesn't have to rebase a branch they're working on with vimdiff :fearful:

**Comment by Sereri** [comment]
Should I wait before merging this in or do you want to create a new pull-request for future stuff?

**Comment by baka-kaba** [comment]
Yeah merge as soon as you can - I was thinking I'd just keep opening pull requests for the branch when I have a bunch of stuff, so I don't get too far off the main branch. This is mostly cleaning up so I'm not really adding any new features, but some of it's a pain to rework and I want to avoid merging with the old versions as much as possible
